mmv1/products/alloydb/User.yaml

```diff
@@ -0,0 +1,103 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'User'
+self_link: '{{cluster}}/users/{{user_id}}'
+base_url: '{{cluster}}/users'
+create_url: '{{cluster}}/users?userId={{user_id}}'
+update_verb: :PATCH
+update_mask: true
+description: 'An AlloyDB User.'
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'AlloyDB': 'https://cloud.google.com/alloydb/docs/'
+  api: 'https://cloud.google.com/alloydb/docs/reference/rest/v1/projects.locations.clusters.users/create'
+async: !ruby/object:Provider::Terraform::PollAsync
+  check_response_func_existence: transport_tpg.PollCheckForExistence
+  target_occurrences: 10
+  actions: ['create', 'update', 'delete']
+import_format: ['{{cluster}}/users/{{user_id}}']
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  custom_import: templates/terraform/custom_import/alloydb_user.go.erb
+autogen_async: true
+skip_sweeper: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'alloydb_user'
+    primary_resource_id: 'default'
+      # Fine-grained resource need different autogenerated tests, as
+      # we need to check destroy during a test step where the parent resource
+      # still exists, rather than during CheckDestroy (when read returns
+      # nothing because the parent resource has then also been destroyed)
+    skip_test: true
+    vars:
+      alloydb_user_id: 'me'
+      alloydb_user_password: 'changeme'
+      alloydb_cluster_id: 'alloydb-cluster'
+      alloydb_instance_id: 'alloydb-instance'
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'alloydb_user_iam_user'
+    primary_resource_id: 'default'
+      # Fine-grained resource need different autogenerated tests, as
+      # we need to check destroy during a test step where the parent resource
+      # still exists, rather than during CheckDestroy (when read returns
+      # nothing because the parent resource has then also been destroyed)
+    skip_test: true
+    vars:
+      alloydb_user_id: 'me@example.com'
+      alloydb_cluster_id: 'alloydb-cluster-iam'
+      alloydb_instance_id: 'alloydb-instance-iam'
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'userId'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The ID of the alloydb user.
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'cluster'
```

Comment:
Is it preferable to have this set up as a single parent resource rather than a series of project/location/cluster?

Generally we have each broken out into an individual field.

cc @GauravJain21

Reply:
That would mean that the user has to set it's location and I'd argue this resource isn't a regional object on it's own but rather part of a regional object.

As the cluster dictates what region the user resides in I wouldn't give the user the option to specify location.

Reply:
True. On the other hand existing resources like `google_sql_user` work this way, where the project and instance are both specified independently: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_user#project

Reply:
The biggest difference here is that `google_sql_user` doesn't rely on a region in his end-point nor is it a field that can be set.
It is implied that it lives in the region where the instance resides in.

`POST https://sqladmin.googleapis.com/v1/projects/{project}/instances/{instance}/users`

I am curious what others believe the right approach is here.